google/cloud/testing_util/validate_metadata.cc

```diff
@@ -68,9 +69,15 @@ RoutingHeaders ExtractMDFromHeader(std::string header) {
   return res;
 }
 
-MATCHER_P(ContainsStdRegex, pattern, "ContainsRegex using std::regex") {
-  std::regex regex(pattern);
-  return std::regex_search(arg, regex);
+/**
+ * We do not use `::testing::MatchesRegex` because our Windows builds use a
+ * googletest built with `GTEST_USES_SIMPLE_RE`, instead of
+ * `GTEST_USES_POSIX_RE`.
+ */
```

Comment:
I would still suggest not having this as a main/doxygen-level comment on `MatchesGlob`.  The decision not to use `testing::MatchesRegex` is only an artifact of how the implementation converts globs to regexs.  That is, move it to the body.

Reply:
So is there is a clean way for me to convert the glob into a form that is accepted by `::testing::MatchesRegex` on all platforms? (clean meaning no `#ifdef`s)

if so, let's use that.

if not, I think the comment makes sense where it is.

Reply:
> is there a clean way for me to convert the glob

I don't think so.  glob `*` conversion requires character classes, which are not supported by `GTEST_USES_SIMPLE_RE`.

> I think the comment makes sense where it is.

I still disagree, no matter what the details of the internal glob->regex conversion.  But, if I haven't convinced you yet, so be it.